File: app/src/main/java/com/projecthub/core/repositories/jpa/ProjectJpaRepository.java
package com.projecthub.core.repositories.jpa;

import com.projecthub.core.models.Project;
import org.springframework.context.annotation.Profile;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

/**
 * Repository interface for {@link Project} entities.
 */
@Repository("jpaProjectRepository")
@Profile("jpa")
public interface ProjectJpaRepository extends JpaRepository<Project, UUID> {

    /**
     * Finds all projects by team ID.
     *
     * @param teamId the UUID of the team
     * @return a list of {@code Project} objects belonging to the team
     */
    List<Project> findAllByTeamId(UUID teamId);

    /**
     * Finds a project with its components by project ID.
     *
     * @param projectId the UUID of the project
     * @return an {@code Optional} containing the project with its components if found
     */
    @NonNull
    @EntityGraph(attributePaths = {"components"})
    Optional<Project> findById(@NonNull UUID projectId);
}

File: src/main/kotlin/com/projecthub/infrastructure/persistence/project/jpa/ProjectJpaRepository.kt
package com.projecthub.infrastructure.persistence.project.jpa

import java.util.UUID
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

/**
 * Spring Data JPA repository for Project entities.
 * This is a pure infrastructure concern and has no knowledge of our domain models.
 */
@Repository
interface ProjectJpaRepository : JpaRepository<ProjectJpaEntity, UUID> {

    /**
     * Finds all projects with the specified status.
     *
     * @param status The status to filter by
     * @return A list of project JPA entities with the given status
     */
    fun findByStatus(status: ProjectJpaEntity.ProjectStatusJpa): List<ProjectJpaEntity>

    /**
     * Finds all projects owned by the specified user.
     *
     * @param ownerId The ID of the owner to filter by
     * @return A list of project JPA entities with the given owner
     */
    fun findByOwnerId(ownerId: UUID): List<ProjectJpaEntity>
}
